activerecord/lib/active_record/relation/calculations.rb

```diff
@@ -319,7 +319,7 @@ def ids
         return relation.ids
       end
 
-      columns = arel_columns([primary_key])
+      columns = arel_columns(Array(primary_key))
```

Comment:
This change wasn't necessary. At least it won't break added test cases. However it is still seems incorrect to call `arel_columns` like `arel_columns([["from", "to"]])` (nested array) as it does nothing and simply returns the passed value:
https://github.com/rails/rails/blob/121712770326b9bca86022992d72811dadb1e133/activerecord/lib/active_record/relation/query_methods.rb#L1719-L1720

Compared to `arel_columns(["from", "to"])` which returns an Array or `Arel` attributes. Though I couldn't come up with a test case to show the need for a change. 

Basically by not making this change we will loose the benefits which are provided by the `arel_column` method:
https://github.com/rails/rails/blob/121712770326b9bca86022992d72811dadb1e133/activerecord/lib/active_record/relation/query_methods.rb#L1725-L1739

But I can't tell which one we could test to show the difference. I don't think aliases handling applies to the primary key case but perhaps we could show the difference by using `TravelRoute.primary_key = ["travel_routes.from", "travel_routes.to"]` which doesn't look like a very realistic use-case

Reply:
I think it's better to maintain the current "internal" behaviour which is to return an `Array` of `Arel` attributes, ie. what you've done.

We can revisit whether it's needed at all in a follow up, but for now, despite it not explicitly affecting our test, changing the intended usage of `arel_columns` seems unwanted.